Tests/Configuration/ConfigTests.cs

```diff
@@ -37,6 +36,20 @@ public void SetRespectsEnvironment()
             Assert.AreNotEqual(betaMode, betaMode2);
         }
 
+        [Test]
+        public void SetNestedObject()
```

Comment:
I think we'll want to include more tests, minimally testing both values of the `isEnvironmentContext` flag, but my vote would be to remove the flag or provide reasoning as to why we need to forms of using `Config.Set` -- unless I'm missing something, I think it can simply be removed.

Perhaps there is special handling regarding environments? My understanding was that we handled it just the same as nested objects, such as `environments.back-testing.my-setting`

Reply:
Thank you  @mchandschuh  for taking time to review all this. Well I agree isEnvironmentContext could be redundant. May I please clarify one thing..  As I could judge by the code of initial Config.Set it would expect that any assignment that has dot separator in its key specifier, i.e nested object, to have relation to one or another environment or not? Simply saying assigment to **environments.back-testing.my-setting** would be done by calling Config.Set(back-testing.my-setting, value) ?

Reply:
I made a small test for initial Set. How would it behave if we want to assign a value to environment that config.json has to information of?  Example:
Config.Set("mega-backtesting.setup-handler", "3285-testing");
I realized it will throw an exception. Here come a commit next .